api/edge/v1alpha1/types.go

```diff
@@ -171,6 +173,129 @@ type PlacementList struct {
+	Destinations []Destination `json:"destinations,omitempty"`
+}
+
+// DownsyncObjectReferences explicitly defines the objects to be down-synced.
+// The ClusterScope list defines the cluster-scope objects, while NamespacedObjects packs individual objects
+// identifiable by namespace & name.
+type DownsyncObjectReferences struct {
+	// `clusterScope` holds a list of individual cluster-scoped objects
+	// to downsync, organized by resource.
+	// Remember that a "resource" is a kind/type/sort of objects,
+	// not an individual object.
+	// +optional
+	ClusterScope []ClusterScopeDownsyncObjects `json:"clusterScope,omitempty"`
+
+	// `NamespaceScope` matches if and only if at least one member matches.
+	// +optional
+	NamespaceScope []NamespaceScopeDownsyncObjects `json:"namespaceScope,omitempty"`
+}
+
+// NamespaceScopeDownsyncObjects matches some objects of one particular namespaced object.
+type NamespaceScopeDownsyncObjects struct {
+	// GroupVersionResource holds the API group, API version and resource name.
+	metav1.GroupVersionResource `json:",inline"`
+
+	// `objectsByNamespace` matches by namespace and name.
+	// An object matches the list if and only if the object matches at least one member of the list.
+	// Thus, no object matches the empty list.
+	// +optional
+	ObjectsByNamespace []NamespaceAndNames `json:"objectsByNamespace,omitempty"`
+}
+
+// NamespaceAndNames identifies some objects of an implied resource that is namespaced.
+// The objects are all in the same namespace.
+type NamespaceAndNames struct {
+	// `namespace` identifies the namespace
+	Namespace string `json:"namespace"`
+
+	// `names` holds the names of the objects that match.
+	// Empty list means none of them.
+	// +optional
+	Names []string `json:"names,omitempty"`
+}
+
+type ClusterScopeDownsyncObjects struct {
+	// GroupVersionResource holds the API group, API version and resource name.
+	metav1.GroupVersionResource `json:",inline"`
+
+	// `objectNames` holds the names of the objects of this kind to downsync.
+	// Empty list means none of them.
+	// +optional
+	ObjectNames []string `json:"objectNames,omitempty"`
+}
+
+// Destination wraps the identifiers required to uniquely identify a destination cluster.
+type Destination struct {
+	ClusterId string `json:"clusterId,omitempty"`
+}
+
+type PlacementDecisionStatus struct {
+	// observedGeneration represents the .metadata.generation that the status is based upon.
+	// For instance, if .metadata.generation is currently 12, but the .status.observedGeneration is 9, the status is out of date
+	// with respect to the current state of the PlacementDecision instance.
+	// Zero means that no status has yet been written here.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// `propagatedWorkloadsCount` represents the number of different destinations that got a copy of the wrapped object
```

Comment:
_the_ wrapped object? I thought the current code (pre-pluggable-transport) is wrapping each workload object individually and that the switch to pluggable transport would preserve that behavior. Certainly the API should not preclude that behavior.

Reply:
the pluggable-transport do not wrap each workload object individually.
I think you were out of office when we discussed transport with Gosia and the others.
we put all objects that are selected by a placement into a single wrapped object.
we also discussed that with Paolo.

Reply:
@nirrozenbaum - @MikeSpreitzer was talking about the pre-pluggable-transport. 
So the answer is - yes the current  pre-pluggable-transport code wrap each object. As part of the pluggable transport mode the team suggested to move to a different approach of wrapping all object in the placement into single wrapping object. There are pros/cons for each of the approaches,  It was agreed to continue with this new approach the team started:
1. The specific approach doesn't change the user facing API
2. When going with the new approach all current functionality need to be supported (The team need to validate)
3. @pdettori had some concerns - the team addressed the issues he raised and he approved
4. changing back to the original approach is estimated as small effort (up to a week) - so we can always decide to switch back if we encounter major reason in the future.

Reply:
Did that discussion include consideration of overlapping Placement objects and the consequent duplication in wrapped bundles?

Reply:
Defining a status field that only makes sense in one approach precludes the other.

Reply:
yes. the discussion included overlapping placement objects and duplication of wrapped bundles.
there is no issue with that.
I didn't understand why you think the status field doesn't make sense.
transport controller watches CR with generation 'x' and creates a manifest work in destination mailbox ns.
we need to be sure that the manifest work content matches whatever there is in our CR with generation 'x'.

Reply:
The go comment here refers to "the [thus, one] wrapped object that contains all the objects in the" workload. That asserts one approach and precludes the other.

Reply:
remove status from this PR.
we'll leave it for next PRs